### PR TITLE
GA TTLAfterFinish

### PR DESCRIPTION
**Pull request by sahilvv**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Graduates TTLAfterFinished feature to GA.

#### Which issue(s) this PR fixes:
Part of kubernetes/enhancements#592

#### Does this PR introduce a user-facing change?
```release-note
TTLAfterFinished is now GA and enabled by default
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/592-ttl-after-finish
https://github.com/kubernetes/enhancements/pull/2841

### CODE REVIEW

**Comment by janetkuo** [comment]
@ahg-g

**Comment by ahg-g** [comment]
can you please update the comments:

1) https://github.com/kubernetes/kubernetes/blob/86d23cf441d2c9f228e9d01bf82deda9eb5c567b/pkg/apis/batch/types.go#L176

2) https://github.com/kubernetes/kubernetes/blob/86d23cf441d2c9f228e9d01bf82deda9eb5c567b/staging/src/k8s.io/api/batch/v1/types.go#L157

**Comment by sahilvv** [comment]
@dims

**Comment by sahilvv** [comment]
@janetkuo @soltysh @liggitt friendly reminder for review. Thanks.

**Review by janetkuo — APPROVED** [review]

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -789,7 +789,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	NodeLease:                                      {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 	NetworkPolicyEndPort:                           {Default: true, PreRelease: featuregate.Beta},
 	ProcMountType:                                  {Default: false, PreRelease: featuregate.Alpha},
-	TTLAfterFinished:                               {Default: true, PreRelease: featuregate.Beta},
+	TTLAfterFinished:                               {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.25
```
Comment (msau42) [inline pkg/features/kube_features.go:792]: Can you also remove all feature gate checks in the controllers?

**Comment by msau42** [comment]
API looks fine.

Please remove the feature tag so the e2e tests run by default: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/ttl_after_finished.go#L43

**Comment by liggitt** [comment]
thanks, API changes approved; please tag when the e2e test and feature gate change is made

**Comment by sahilvv** [comment]
@msau42 @liggitt addressed the review feedback for removing featureGate from controllers and fixing the e2e test. Please review.

**Pull request merged**